src/sinol_make/helpers/compile.py

```diff
@@ -1,15 +1,59 @@
-from typing import Tuple, Union
+from typing import Tuple, Union, Dict
 import os
 import sys
 import shutil
 import stat
 import subprocess
 
 import sinol_make.helpers.compiler as compiler
+from sinol_make import util
 from sinol_make.interfaces.Errors import CompilationError
 from sinol_make.structs.compiler_structs import Compilers
 
 
+def get_executable_info_file(file_path):
+    """
+    Calculate the md5 sum of file's content and return the path to file `cache/md5sums/<md5sum>`.
+    If this file exists it contains the path to the compiled executable.
```

Comment:
```suggestion
    If this file exists it contains the path to the compiled executable.
    Thanks to that, we cache the compiled solutions and recompile them when they change.
```

Reply:
It's always useful to provide *context* behind the code. In this case: why do we care about the path in a file, or why do we compute the md5sums?

It's best when the comments explain the *intention*, not *implementation details*.